src/FundSourceApprover/FundSourceAllower/Erc20FundSourceAllower.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+import "./FundSourceAllowerBase.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @notice Funds allower for an Erc20 token
+contract Erc20FundSourceAllower is FundSourceAllowerBase {
+    using SafeERC20 for IERC20;
+
+    // Erc20 token to receieve / transfer to destination
+    IERC20 public immutable token;
```

Comment:
risk of fund loss if other token or eth is sent here

Reply:
no recieve/fallback for eth 🤷. for tokens, these are meant to be used by a single orbit chain, so I'm not too concerned about confused user error. we could add DAO recovery but I think I'd prefer to avoid adding more stuff?